src/main/java/Duke.java

```diff
@@ -78,39 +40,109 @@ public static void markTask(Task[] taskList, String markAction, int taskNumber)
+     * @return array of Strings (task type, description, additional info)
+     */
+    public static String[] extractTaskInfo(String inputLine) {
+        String type;
+        String description;
+        String additionalInfo;
+        int additionalInfoIndex = inputLine.length(); // Index where description should end
+
+        // Extract the type of task
+        type = inputLine.split(" ")[0];
+
+        // Extract the deadline (for Deadline tasks) or time (for Event tasks), if applicable
+        if (type.equals("deadline")) {
+            additionalInfo = inputLine.substring(inputLine.indexOf("/by") + 4);
+            additionalInfoIndex = inputLine.indexOf("/by");
+        } else if (type.equals("event")) {
+            additionalInfo = inputLine.substring(inputLine.indexOf("/at") + 4);
+            additionalInfoIndex = inputLine.indexOf("/at");
+        } else {
+            additionalInfo = "";
+        }
+
+        // Extract the description
+        if (type.equals("todo")) {
+            description = inputLine.substring(type.length() + 1);
+        } else if (type.equals("deadline") || type.equals("event")) {
+            description = inputLine.substring(type.length() + 1, additionalInfoIndex - 1);
+        } else { // No task type specified; the entire line is the task description
+            description = inputLine;
+        }
+
+        return new String[] { type, description, additionalInfo };
+    }
+
+    /**
+     * Helper for processTasks
+     * Converts line of user input into relevant Task object
+     * Adds Task object to global taskList
+     * @param line line of user input
+     */
+    public static void addTask(String line) {
+        String[] taskInfo = extractTaskInfo(line);
+        String type = taskInfo[0];
+        String description = taskInfo[1];
+        String additionalInfo = taskInfo[2];
+        Task task = new Task(line);
+
+        if (type.equals("todo")) {
+            task = new Todo(description);
+        } else if (type.equals("deadline")) {
+            task = new Deadline(description, additionalInfo);
+        } else if (type.equals("event")) {
+            task = new Event(description, additionalInfo);
+        }
+
+        taskList[taskIndex] = task;
+        taskIndex += 1;
+        System.out.println("Got it. I've added this task:");
+        System.out.println(task.toString());
+        System.out.println("Now you have " + Integer.toString(taskIndex) + " tasks on the list.");
+    }
+
+    /**
+     * Processes user input (addition of tasks to the global taskList) until user types "bye"
+     */
+    public static void processTasks() {
         String line;
         Scanner in = new Scanner(System.in);
         line = in.nextLine();
+        // While the user has not exited the program
         while (!line.equals("bye")) {
             System.out.println(division);
+            // List out the tasks
             if (line.equals("list")) {
                 printTaskObjects(taskList);
```

Comment:
According to the coding standard, it's better to have the respective comments under "while" and "if". 
An eg taken from Java coding standard,
while (true) {
    // Do something
    something();
}